fapi-client/src/main/scala/com/gu/facia/api/models/collection.scala

```diff
@@ -152,17 +154,29 @@ object Collection extends StrictLogging {
       Response.Async.Right(futureMaybeAtomData)
     }
 
-    def isValidMediaAtom(atom: Atom, id: String): Boolean = {
+    //
+    // We need to make sure that we only select the main media atom rather than another embedded atom. This follows the same pattern implemented in Frontend.
+    //
+    def extractMainMediaAtomIdFromHtml(html: String): Option[String] = {
```

Comment:
I'm not sure why you need to parse the html to get the main media atom.
The CAPI response has this in response->content->blocks->main->elements->contentAtomTypeData->atomId